tests/cpp_unit_tests/test_optimizer.hpp

```diff
@@ -165,6 +167,48 @@ constexpr auto strategies_and_methods = [] {
     }
     return result;
 }();
+
+struct OptimizerStrategySide {
+    OptimizerStrategy strategy{};
+    ControlSide side{};
+};
+
+constexpr auto strategies_and_sides = [] {
+    std::array<OptimizerStrategySide, strategies.size() * tap_sides.size()> result;
+    size_t idx{};
+    for (auto strategy : strategies) {
+        for (auto side : tap_sides) {
+            result[idx++] = {strategy, side};
+        }
+    }
+    return result;
+}();
+
+struct OptimizerStrategySearchSide {
+    OptimizerStrategy strategy{};
+    SearchMethod search{};
+    ControlSide side{};
+};
+
+constexpr auto search_methods = [] { return std::array{SearchMethod::scanline, SearchMethod::binary_search}; }();
+
+constexpr auto strategy_search_and_sides = [] {
```

Comment:
this kind of combinatoric expansion is something doctest is really bad at. if you prefer you can even go as far as to combine them into 1 array (maybe even with `strategies_and_methods`) cause i see that you have to use it quite often.

Reply:
I had one attempt in https://github.com/PowerGridModel/power-grid-model/pull/668/commits/6c0282c0eb6570209522786c8f540e56346e0142 but sonar cloud kept complaining. This is the resulted compromise following existing scheme.

Reply:
ok :+1: